src/components/results/common/global-filter/results-global-filter.tsx

```diff
@@ -0,0 +1,397 @@
+                case FilterType.FILTER:
+                    return option.label;
+            }
+        },
+        [translate]
+    );
+
+    const handleChange = useCallback(
+        (globalFilters: GlobalFilter[]): void => {
+            let globalFiltersToAddToRecents: GlobalFilter[] = globalFilters;
+            Promise.all(
+                // checks if the generic filters still exist, and sets their path value
+                globalFilters
+                    .filter((globalFilter: GlobalFilter) => globalFilter.filterType === FilterType.FILTER)
+                    .map(
+                        (fetchedGlobalFilter: GlobalFilter) =>
+                            fetchedGlobalFilter.uuid &&
+                            fetchDirectoryElementPath(fetchedGlobalFilter.uuid)
+                                .then((response: ElementAttributes[]) => {
+                                    const parentDirectoriesNames = response.map((parent) => parent.elementName);
+                                    const path = computeFullPath(parentDirectoriesNames);
+                                    let fetchedFilter: GlobalFilter | undefined = globalFilters.find(
+                                        (globalFilter) => globalFilter.uuid === fetchedGlobalFilter.uuid
+                                    );
+                                    if (fetchedFilter && !fetchedFilter.path) {
+                                        fetchedFilter.path = path;
+                                    }
+                                })
+                                .catch(() => {
+                                    if (fetchedGlobalFilter.uuid) {
+                                        // remove those missing filters from recent global filters
+                                        dispatch(removeFromRecentGlobalFilters(fetchedGlobalFilter.uuid));
+                                        globalFiltersToAddToRecents = [
+                                            ...globalFiltersToAddToRecents.filter(
+                                                (globalFilter) => globalFilter.uuid !== fetchedGlobalFilter.uuid
+                                            ),
+                                        ];
+                                    }
+                                    snackError({
+                                        messageTxt: fetchedGlobalFilter.path,
+                                        headerId: 'ComputationFilterResultsError',
+                                    });
+                                })
+                    )
+            ).then(() => {
+                setSelectedGlobalFilters(globalFilters);
+                // Updates the "recent" filters unless they have not been found
+                dispatch(addToRecentGlobalFilters(globalFiltersToAddToRecents));
+                onChange(globalFilters);
+            });
+        },
+        [dispatch, onChange, snackError]
+    );
```

Comment:
maybe using 'async await' can make code here more readable : 
```suggestion
 const handleChange = useCallback(
  async (globalFilters: GlobalFilter[]): Promise<void> => {
    let validFilters: GlobalFilter[] = [];

    await Promise.all(
      globalFilters
        .filter((filter) => filter.filterType === FilterType.FILTER && filter.uuid)
        .map(async (filter) => {
          try {
            const response = await fetchDirectoryElementPath(filter.uuid!);
            const path = computeFullPath(response.map((parent) => parent.elementName));
            validFilters.push({ ...filter, path });
          } catch (error) {
            dispatch(removeFromRecentGlobalFilters(filter.uuid!));
            snackError({ messageTxt: filter.path, headerId: 'ComputationFilterResultsError' });
          }
        })
    );

    const updatedFilters = globalFilters.map(filter =>
      validFilters.find(validFilter => validFilter.uuid === filter.uuid) || filter
    );

    setSelectedGlobalFilters(updatedFilters);
    dispatch(addToRecentGlobalFilters(validFilters));
    onChange(updatedFilters);
  },
  [dispatch, onChange, snackError]
);
```

Reply:
Indeed this is (a bit) more readable. Especially `await Promise.all(fetchFiltersPromises);`

Updated.